tests/proxy/replicated_map/replicated_map_test.go

```diff
@@ -0,0 +1,323 @@
+	. "github.com/hazelcast/hazelcast-go-client/tests"
+	"log"
+	"reflect"
+	"sort"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+)
+
+var rmp core.ReplicatedMap
+var client hazelcast.IHazelcastInstance
+
+func TestMain(m *testing.M) {
+	remoteController, err := NewRemoteControllerClient("localhost:9701")
+	if remoteController == nil || err != nil {
+		log.Fatal("create remote controller failed:", err)
+	}
+	cluster, err := remoteController.CreateCluster("3.9", DEFAULT_XML_CONFIG)
+	remoteController.StartMember(cluster.ID)
+	client, _ = hazelcast.NewHazelcastClient()
+	rmp, _ = client.GetReplicatedMap("myReplicatedMap")
+	m.Run()
+	rmp.Clear()
+	client.Shutdown()
+	remoteController.ShutdownCluster(cluster.ID)
+}
+
+func TestReplicatedMapProxy_Destroy(t *testing.T) {
+	testKey := "testingKey"
+	testValue := "testingValue"
+	rmp.Put(testKey, testValue)
+	rmp.Destroy()
+	rmp, _ := client.GetReplicatedMap("myReplicatedMap")
+	res, err := rmp.Get(testKey)
+	AssertNilf(t, err, res, "replicatedMap Destroy() failed")
+}
+
+func TestReplicatedMapProxy_Put(t *testing.T) {
+	defer rmp.Clear()
+	testKey := "testingKey"
+	testValue := "testingValue"
+	newValue := "newValue"
+	_, err := rmp.Put(testKey, testValue)
+	res, err := rmp.Get(testKey)
+	AssertEqualf(t, err, res, testValue, "replicatedMap Put() failed")
+	oldValue, err := rmp.Put(testKey, newValue)
+	AssertEqualf(t, err, oldValue, testValue, "replicatedMap Put()  failed")
+}
+
+func TestReplicatedMapProxy_PutWithTtl(t *testing.T) {
+	defer rmp.Clear()
+	testKey := "testingKey"
+	testValue := "testingValue"
+	rmp.Put(testKey, testValue)
+	oldValue, err := rmp.PutWithTtl(testKey, "nextValue", 100, time.Second)
+	AssertEqualf(t, err, oldValue, testValue, "replicatedMap PutWithTtl()  failed")
+	res, err := rmp.Get(testKey)
+	AssertEqualf(t, err, res, "nextValue", "replicatedMap PutWithTtl() failed")
+}
+
+func TestMapProxy_PutWithTtlWhenExpire(t *testing.T) {
+	defer rmp.Clear()
+	testKey := "testingKey"
+	testValue := "testingValue"
+	rmp.Put(testKey, testValue)
+	rmp.PutWithTtl(testKey, "nextValue", 1, time.Millisecond)
+	time.Sleep(2 * time.Second)
+	res, err := rmp.Get(testKey)
+	AssertNilf(t, err, res, "replicatedMap PutWithTtl() failed")
+}
+
+func TestReplicatedMapProxy_PutAll(t *testing.T) {
+	defer rmp.Clear()
+	testMap := make(map[interface{}]interface{})
+	for i := 0; i < 10; i++ {
+		testMap["testingKey"+strconv.Itoa(i)] = "testingValue" + strconv.Itoa(i)
+	}
+	err := rmp.PutAll(testMap)
+	AssertErrorNil(t, err)
+	entryList, err := rmp.EntrySet()
+	AssertMapEqualPairSlice(t, err, testMap, entryList, "replicatedMap PutAll() failed")
+}
+
+func TestReplicatedMapProxy_Get(t *testing.T) {
+	defer rmp.Clear()
+	testKey := "testingKey"
+	testValue := "testingValue"
+	_, err := rmp.Put(testKey, testValue)
+	res, err := rmp.Get(testKey)
+	AssertEqualf(t, err, res, testValue, "replicatedMap Get() failed")
+}
+
+func TestReplicatedMapProxy_ContainsKey(t *testing.T) {
```

Comment:
We also need test with different values. The methods that take values should be tested for both exist and not exists  cases.
Test Contain for an empty map and/or Remove a non exists key.
See queue_test.go